core/src/main/scala/kafka/tools/JmxTool.scala

```diff
@@ -80,8 +83,32 @@ object JmxTool extends Logging {
     val attributesWhitelist = if(attributesWhitelistExists) Some(options.valueOf(attributesOpt).split(",")) else None
     val dateFormatExists = options.has(dateFormatOpt)
     val dateFormat = if(dateFormatExists) Some(new SimpleDateFormat(options.valueOf(dateFormatOpt))) else None
-    val jmxc = JMXConnectorFactory.connect(url, null)
-    val mbsc = jmxc.getMBeanServerConnection()
+
+    var jmxc : JMXConnector = null
+    var mbsc : MBeanServerConnection = null
+    var retries = 0
+    val maxNumRetries = 20
+    var connected = false
+    while (retries < maxNumRetries && !connected) {
+      try {
+        System.err.println("Trying to connect to JMX url: %s".format(url))
+        jmxc = JMXConnectorFactory.connect(url, null)
+        mbsc = jmxc.getMBeanServerConnection()
+        connected = true
+      } catch {
+        case e : Exception => {
+          System.err.println("Could not connect to JMX url: %s. Exception %s".format(url, e.getMessage))
+          retries += 1
+          Thread.sleep(500)
+        }
+      }
+    }
+
+    if (!connected) {
+      System.err.println("Could not connect to JMX url %s after %d retries".format(url, maxNumRetries))
+      System.err.println("Exiting")
+      System.exit(1)
```

Comment:
`sys.exit` is slightly better as the compiler knows that the call never returns.